optimum/habana/diffusers/models/attention_processor.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import os
+from typing import Optional, Union
+
+import torch
+import torch.nn.functional as F
+from diffusers.models.attention_processor import Attention
+from diffusers.utils import USE_PEFT_BACKEND, logging
+from diffusers.utils.import_utils import is_xformers_available
+from torch import nn
+
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+if is_xformers_available():
+    import xformers
+    import xformers.ops
+else:
+    xformers = None
+
+
+class Softmax(nn.Module):
+      def __init__(self):
+          super().__init__()
+
+      def forward(self, x, dim = None, invAttnHead= None):
+          return torch.ops.hpu.softmax_fp8(x, dim,  None, None, invAttnHead)
```

Comment:
why is softmax_fp8 used here? support stable diffuser for fp8?

Reply:
Current support is for FP8 only.